packages/react/vite.config.ts

```diff
@@ -1,39 +1,31 @@
 import { defineConfig, type PluginOption } from 'vite'
 import react from '@vitejs/plugin-react'
+import gzipPlugin from 'rollup-plugin-gzip';
 import { visualizer } from 'rollup-plugin-visualizer';
 
 import BOSConfig from './bos.json' assert { type: 'json' };
 
-const { environments, filePathMatchers } = BOSConfig
+const { environments } = BOSConfig
 const { deployerAccount, fileServerUrl } = environments.testnet;
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [react(),visualizer() as PluginOption, {
+  plugins: [react(),visualizer() as PluginOption, gzipPlugin(), {
     name: 'inject-bundle-url',
     apply: 'build',
     transformIndexHtml(html ) {
       return html.replace(/(href|src)="\/assets\//g, `$1="${fileServerUrl}/${deployerAccount}/`);
     }
   }],
+  define: {
+    global: {},
+    process: { env: {} }
+  },
   build: {
     rollupOptions: {
       output: {
-        manualChunks: function (id, x) {
-          if (id.includes('BosRoot')) {
-            return 'preset';
-          }
-
-          if (filePathMatchers.some((m) => id.includes(m))) {
-            return 'dev';
-          }
-
-          if (id.match(/\/node_modules\/[^/]+\/react/g)) {
-            const depRootIndex = id.indexOf('/react');
-            if (depRootIndex > -1) {
-              return 'react';
-            }
-          }
+        manualChunks: {
+          preset: ['@chain-deployed-ui/react-preset', 'react', 'react-dom']
```

Comment:
@mpeterdev including a fixed set of packages in the preset bundle is easier than I thought 😄

Reply:
awesome!